module/VuFindSearch/src/VuFindSearch/ParamBag.php

```diff
@@ -163,11 +163,17 @@ public function add($name, $value, $deduplicate = true)
             $this->params[$name] = [];
         }
         if (is_array($value)) {
-            $this->params[$name] = array_merge($this->params[$name], $value);
+            $this->params[$name] = array_merge_recursive($this->params[$name], $value);
         } else {
             $this->params[$name][] = $value;
         }
         if ($deduplicate) {
+            // Avoid deduplicating associative array params (like Primo filterList):
+            foreach ($this->params[$name] as $key => $current) {
```

Comment:
Would it make sense to use [array_is_list](https://www.php.net/manual/en/function.array-is-list.php) here, or are you concerned about supporting non-consecutive numeric-keyed arrays?

Reply:
I think we can deduplicate and do array_values() for arrays with non-consecutive numeric keys, so I decided to allow those too.

Reply:
Not sure if that's a case ever encountered, though, so if you'd prefer array_is_list, I'd be fine with that too.

Reply:
No, this is fine as-is, just wanted to check -- always fun to use a shiny new function when it applies. ;-)